python/tk_multi_publish2/api/plugins/instance_base.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2018 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import sgtk
+from .setting import Setting
+
+logger = sgtk.platform.get_logger(__name__)
+
+
+class PluginInstanceBase(object):
+    """
+    A base class for functionality common to plugin hooks (collectors and
+    publish plugins).
+
+    Each object reflects an instance in the app configuration.
+    """
```

Comment:
Could you add a serialized example?

Reply:
same comment about making the serialized state public.